### ISSUE TITLE
Conditional behavior in debug/release mode for cxx::Expects and cxx::Ensures 

### ISSUE THREAD

**Issue by marthtz** [eclipse-iceoryx/iceoryx#607]
## Brief feature description

Add conditional behavior on debug/release mode for `cxx::Expects` and `cxx::Ensures`.

## Detailed information

Current state is that `cxx::Expects` and `cxx::Ensures` terminate if the condition is not met.
Suggestion to add conditional behavior on debug/release mode for both functions, like in debug mode you will be terminated with a detailed error message and in release mode we continue.

Optional/to be discussed:
- Add a parameter allowing continuation, e. g.:
   ```
   cxx::Expects( myNumber > 0 ); // hard, calls std::terminate always
   cxx::Expects( myNumber > 0, ALGORITHM_CAN_CONTINUE); // we can continue even the requirement is not fulfilled
   ```
- Can there be a `constexpr` version of `cxx::Expects` and `cxx::Ensures`?

**Comment by elBoberido** [comment]
I don't think this is the right path forward. In your example you don't need the `Expects` because you already decided you can handle the error. `Expects` is for invariants which should never be broken but your inner chicken is appeased by this extra check. Please don't use constructs for problems they are not designed for.
